### DISCUSSION TITLE
Divers problèmes - GRR V4.3.1

### DISCUSSION THREAD

**Discussion by acdct** [JeromeDevome/GRR#341] [Q&A]
Bonjour,

J'ai des soucis sur la version 4.3.1 que je viens d'installer.

- J'ai créé un nouveau site, en voulant le supprimer j'ai le message d'erreur : _Truncated incorrect DOUBLE value: '' Q=update grr_utilisateurs set default_site = '-1' where default_site=''_

- Dans la configuration générale > Apparence :

1. impossible de modifier l'affichage des listes des domaines et ressources
2. les listes déroulante "Site par défaut" et "Domaine par défaut" sont vides malgré l'existence de données.
3. (Testé sur une installation locale) : en modifiant le domaine par défaut, message d'erreur : _Incorrect integer value: '' for column 'default_area' at row 1 Q=UPDATE grr_utilisateurs SET default_area=''_
4. impossible de modifier le choix du thème. Malgré le message synchronisation terminée, le choix à la donnée initiale (c'était à Grand bleu, je modifie à Verdoyant mais ça revient à Grand bleu)
5. impossible de modifier l'affichage de la ressource (journée, semaine, etc)
6. impossible de modifier la langue également

Merci d'avance pour votre aide.

Cordialement

**Comment by ynaessens** [comment]
Bonjour,
mes constats :
- création de site, OK, suppression de site KO
- modification de l'affichage des listes : OK. Il faut garder présent à l'esprit que les modifications ne s'appliquent que pour les utilisateurs ayant le réglage par défaut. Dans le cas d'une synchronisation des préférences, la modification ne s'applique qu'à la prochaine session.
- listes de sites ou domaines : vides, KO. De ce fait, je me demande comment vous avez pu modifier le domaine par défaut.
- pour le thème, c'est comme pour les listes. La préférence individuelle l'emporte sur la préférence par défaut et les modifications globales ne s'appliquent que lors de la prochaine session. OK.
- pour l'affichage des plannings (semaine, journée, etc.), c'est toujours le même principe. OK.
- pour la langue, le principe est le même, mais effectivement il y a un dysfonctionnement lorsque le planning s'affiche en visiteur non connecté. KO.

Merci pour votre signalement. J'espère que mes remarques vous auront aidé.
Cordialement,
YN

**Reply by acdct** [reply]
Bonjour,
Merci pour vos retours.

Pour être plus précis, sur une **nouvelle instance que j'ai installé en local**, au niveau de l'administration :
- Je n'ai pas l'onglet "Sites" dans le menu de gauche, j'ai peut-être raté quelque chose.
- Dans Apparence :
--- J'ai bien le domaine que j'ai créé, mais en le choisissant : _Incorrect integer value: '' for column 'default_area' at row 1 Q=UPDATE grr_utilisateurs SET default_area=''_
--- En choisissant le type d'affichage des ressources : _Incorrect integer value: '' for column 'default_area' at row 1 Q=UPDATE grr_utilisateurs SET default_area=''_
--- Choix du thème : la liste revient systématiquement sur le choix de départ (je comprend que la préférence individuelle l'emporte, mais le choix de l'admin ne devrait pas resté affiché dans la liste, indépendamment du choix utilisateur ? Car c'est bien le cas sur les anciennes versions de GRR que nous avons actuellement)
--- Choix de la langue : idem que le thème

Sur une **instance mise à jour de v3.4.3 vers 4.3.1**, au niveau de l'administration :
- J'ai bien un onglet "Sites" avec 2 données (avec toujours l'erreur lors de la suppression)
- Dans Apparence :
--- Même problème que sur l'instance locale. Mais cette fois-ci, j'ai bien des sites et des domaines. Cependant, les listes dans "apparence" sont vides et restent à "Choisissez un site" et "Choisissez un domaine"

Merci

**Reply by ynaessens** [reply]
Les sites sont une fonctionnalité à activer. Par défaut il n'y a qu'un site.

**Reply by acdct** [reply]
C'est noté pour les sites, merci.
Cependant le reste est bien un problème applicatif ?

**Reply by JeromeDevome** [reply]
Bonjour, un certains nombre de commits on lieu pour fonctionnalité site. L'ensemble des corrections sera en version 4.3.2 (probablement ce week-end)

**Comment by JeromeDevome** [comment]
Bonjour,
La version [4.3.2](https://github.com/JeromeDevome/GRR/releases/tag/v4.3.2) corrige pas mal de problème sur les sites.

Je vous conseil de faire la mise à jour, et de revenir vers nous avec les problèmes restant ou de nous dire que tout fonctionne :)

Bonne journée.

**Reply by acdct** [reply]
Bonjour,
J'ai installé une version vierge de la 4.3.2. Tous les problèmes indiqués dans mon premier message demeurent malheureusement. J'ai toujours les messages d'erreur ainsi que le non changement des listes lors de mon choix pour le thème ou la langue.
Pour les sites, je ne vois nulle part l'endroit pour les activer.
Merci

**Reply by ynaessens** [reply]
Bonjour,
le choix des thèmes et langues fait au niveau de l'utilisateur est prioritaire sur celui fait par l'administrateur (valeurs par défaut).
L'activation du multisites se fait dans le menu d'administration, Onglet Configuration générale, puis Modules.
Cordialement,
YN

**Reply by acdct** [reply]
Bonjour,

Merci pour les multisites, c'est bien activé.

Pour le choix des thèmes et langues, je comprend que celui de l'utilisateur est prioritaire. Toutefois, lorsque l'administrateur en choisi un par défaut, ce dernier ne doit-il pas resté au niveau de la liste ? J'ai par exemple "Defaut" pour le thème. Je voudrai "Verdoyant", lorsque je choisis verdoyant et que je valide, la liste revient systématiquement à Defaut et ne garde pas en mémoire mon choix. Idem pour la langue et le type d'affichage des domaines et ressources.

Quant au pavé "Site par défaut", "Domaine par défaut" et "Ressources affichés", toujours l'erreur : Incorrect integer value: '' for column 'default_area' at row 1 Q=UPDATE grr_utilisateurs SET default_area=''

Merci pour votre aide précieuse !

Cordialement

**Reply by ynaessens** [reply]
Bonjour,
j'ai perdu de vue cette discussion, désolé.
Avec les derniers scripts de la branche master de GRR, je ne reproduis pas ces erreurs. Pouvez-vous essayer ?
Cordialement,
YN

**Reply by acdct** [reply]
Bonjour,
Il n'y a pas de souci :)
Malheureusement les  mêmes problèmes demeurent avec la V4.3.3 sur une installation vierge.
PHP 8.3.4

**Comment by acdct** [comment]
Bonjour,
Je viens d'installer la 4.3.4 et ça persiste.
Je suis allé voir dans la table grr_setting, est-ce normal que je ne trouve pas **default_area** qui est uniquement dans la table grr_utilisateurs alors qu'il s'agit d'une configuration générale ?
J'essaye de fouiller pour comprendre pourquoi les listes par défaut ne garde pas la donnée.
Merci !

**Comment by ynaessens** [comment]
Bonjour,
je reviens sur cette question d'entrée default_area qui est absente de votre table grr_setting.
Sur mon installation (v4.3.4 mise à jour en 4.3.6), l'entrée est bien présente et mise à jour par la page de configuration générale, onglet Apparence.
La synchronisation des réglages avec tous les utilisateurs ne change pas la valeur.
Quelle est l'historique de votre base de données ?
Cordialement,
YN

**Reply by acdct** [reply]
Bonjour,

Sur un update d'une instance déjà existante, defaut_area existe bien. C'est sur une installation vierge que cette entrée est absente.

Je viens de mettre à jour vers la dernière version. En base, table "grr_settings" j'ai par exemple default_area = 12.

Lorsque je sélectionne un autre domaine dans "Apparence", le nouveau choix n'est pas pris en compte et au niveau de la table, l'entrée reste à 12 alors que j'ai choisi un domaine dont l'ID est 3.

De plus "default_site", "default_area" ont bien une valeur, mais au niveau visuel ça reste à (Choisissez un site) et (Choisissez un domaine).

Cordialement

**Reply by ynaessens** [reply]
Bonjour,
je ne reproduis pas exactement ce que vous décrivez.
La valeur de default_area est bien mise à jour en base de données, ainsi que default_site et default_room (ce dernier paramètre est celui qui gère le type de planning affiché).
Je constate cependant qu'en mode non connecté l'affichage ne tient pas compte de ces paramètres. Il y a là une régression par rapport à la version 3.5.1. Correction à venir...
Cordialement,
YN

**Reply by acdct** [reply]
Bonjour,
Je vous avoue que j'ai aussi du mal à comprendre...
Je vais donc attendre la correction et installer la future version pour voir comment ça se passe.
Encore merci.
Cordialement
